nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -595,41 +655,172 @@ sub multiLineList {
+  # Enable graphical acceleration for NVIDIA GPUs (requires unfree
+  # packages, see above).
+  #
+  # services.xserver.videoDrivers = [ "nvidia" ];
+  #
+  # For older NVIDIA GPUs you can use the following.
+  #
+  # services.xserver.videoDrivers = [ "nvidiaLegacy340" ];
+  # or
+  # services.xserver.videoDrivers = [ "nvidiaLegacy304" ];
+  # or
+  # services.xserver.videoDrivers = [ "nvidiaLegacy173" ];
+  #
+  # Acceleration on modern AMD/ATI and Intel GPUs will work with free
+  # drivers installed out of the box, but you might still want to
+  # override the default value of this option just to save some space
+  # by not installing the drivers you don't need. E.g.
+  #
+  # services.xserver.videoDrivers = [ "intel" ];
+  # or
+  # services.xserver.videoDrivers = [ "ati" ];
+  #
+  # Some older ATI GPUs might need the following to get the
+  # acceleration working (requires unfree packages, see above).
+  #
+  # services.xserver.videoDrivers = [ "ati_unfree" ];
+  # or
+  # services.xserver.videoDrivers = [ "amdgpu" ];
+  # or
+  # services.xserver.videoDrivers = [ "amdgpu-pro" ];
+
+  # Enable support for 32-bit OpenGL libraries on x86_64 which is
+  # useful if you want to run 32-bit Windows games under Wine.
+  #
+  # hardware.opengl.driSupport32Bit = true;
+  # environment.systemPackages = with pkgs; [ wine winetricks ];
+
+  # Some of the drivers above might not work with the current Linux
+  # kernel, so you might end up overriding that too.
+  #
+  # boot.kernelPackages = pkgs.linuxPackages_4_9;
 
   # Define a user account. Don't forget to set a password with ‘passwd’.
-  # users.extraUsers.guest = {
+  #
+  # users.users.me = {
   #   isNormalUser = true;
   #   uid = 1000;
+  #   extraGroups = [ "input" "audio" "video" "cdrom" ]
+  #              ++ [ "wheel" ]; # for sudo
+  #   openssh.authorizedKeys.keyFiles = [ /etc/nixos/id_rsa.pub ];
   # };
 
-  # This value determines the NixOS release with which your system is to be
-  # compatible, in order to avoid breaking some software such as database
-  # servers. You should change this only after NixOS release notes say you
-  # should.
-  system.stateVersion = "${\(qw(@release@))}"; # Did you read the comment?
+  # Enable the KDE Desktop Environment.
+  #
+  # services.xserver.displayManager.sddm.enable = true;
+  # services.xserver.desktopManager.plasma5.enable = true;
+
+  # Or maybe you want to enable Xfce Desktop Environment and Slim
+  # display manager?
+  #
+  # services.xserver.displayManager.slim.enable = true;
+  # services.xserver.desktopManager.xfce.enable = true;
+
+  # Or maybe you just want to automatically login into your .xsession
+  # script? Then don't enable any desktopManagers and uncomment the
+  # following.
+  #
+  # services.xserver.displayManager.auto = {
+  #   enable = true;
+  #   user = "me";
+  # };
+
+  # If you're in a hurry, you may stop editing your configuration.nix now
+  # and completely ignore everything below.
```

Comment:
:-1: this file will be immediately out of date because nothing can update it. We shouldn't be talking about best practices or general techniques. This belongs in other documentation elsewhere. The parts about unfree software are universally true. The remainder of this should definitely not be in this config file, but instead in the documentation or wiki.

Reply:
I agree, this is the main problem with this PR. It's fine to install a small configuration.nix with some options to get people started, but it should not duplicate the manual or try to cover a gazillion use cases.

Reply:
Well, experimental evidence shows that not enough people read the manual anyway…

Reply:
+1. Using a separate user to run e.g. firefox sounds like a great idea, but the boilerplate in configuration.nix is not the soapbox from which to broadcast this position. People approaching this file for the first time just want to get NixOS running, which already has enough mind-blowing paradigm shifts to absorb.

Reply:
agree, :-1: on the somewhat niche and cumbersome desktop-only per-software user approach here. It would be better to a link to general information about running NixOS on the desktop with a section on per-software users.